include/solvers/mpm_base.h

```diff
@@ -24,6 +24,13 @@
 
 namespace mpm {
 
+//! Variable type
+//! Scalar: boolean, unsigned, int, double
+//! Vector: Vector of size 3
+//! Tensor: Symmetric tensor arranged in voigt notation
+enum class VariableType { Scalar, Vector, Tensor };
+extern tsl::robin_map<std::string, VariableType> variables;
```

Comment:
I was thinking of moving it inside the class similar to `vtk_vars_`

Reply:
Which class do you mean here?

Reply:
In to MPMBase, so we can derive them in MPMExplicit.